File: pinyougou_parent/pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/AreasController.java
package com.pinyougou.manager.controller;
import java.util.List;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbAreas;
import com.pinyougou.sellergoods.service.AreasService;
import com.pinyougou.util.ObjectUtils;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/areas")
public class AreasController {

    @Reference
    private AreasService areasService;

    /**
     * 查询所有 (包含模糊查询和分页)
     * @param areas
     * @param pageNum
     * @param pageSize
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(@RequestBody(required = false) TbAreas areas, Integer pageNum, Integer pageSize){
        return areasService.listAreasByExample(areas,pageNum, pageSize);
    }

    /**
     * 删除方法
     * @param idList
     * @return
     */
    @RequestMapping("/removeAreas")
    public Result removeAreas(Integer[] idList) {
        if (!ObjectUtils.notEmpty(idList)) {
            return new Result(false, "未选中要删除的条目");
        }
        try {
            return areasService.removeAreas(idList);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }

    /**
     * 修改对象
     * @param areas
     * @return
     */
    @RequestMapping("/updateAreas")
    public Result updateAreas(@RequestBody(required = false) TbAreas areas) {
        if (!ObjectUtils.notEmpty(areas)) {
            return new Result(false, "对象不存在");
        }
        try {
            return areasService.updateAreas(areas);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }

    /**
     * 根据id查询
     * @return
     */
    @RequestMapping("/findOne")
    public TbAreas findOne(Integer id){
        return areasService.getAreasById(id);
	}
	
    /**
     * 新增数据
     * @param areas
     * @return
     */
    @RequestMapping("/saveAreas")
    public Result saveAreas(@RequestBody(required = false) TbAreas areas){
        if (!ObjectUtils.notEmpty(areas)) {
            return new Result(false, "对象不存在");
        }
        try {
            return areasService.saveAreas(areas);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "未知的错误信息,请重试");
        }
    }
    /**
     * 返回全部列表(已过时的列表查询方法)
     * @return
     */
    @RequestMapping("/findAll")
    public List<TbAreas> findAll(){
        return areasService.findAll();
    }
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/controller/areasController.js
app.controller('areasController', function ($scope, $controller, areasService) {//控制器
    //引入父类控制器
    $controller('baseController', {$scope: $scope});
    /**
     * 发送查询请求,定义findPage方法(已过时)
     */
    $scope.findPage = function (pageNum, pageSize) {
        areasService.findPage(pageNum, pageSize).success(function (data) {
					  
            $scope.rows = data.rows;
            $scope.paginationConf.totalItems = data.total;//更新总记录数
        })
    };
	
    /**
     * 搜索 (升级版findPage)
	 * 读取列表数据绑定到表单中
     */
    $scope.search = function (pageNum, pageSize) {
        areasService.search(pageNum, pageSize, $scope.searchEntity).success(function (data) {
					  
            $scope.rows = data.rows;
            $scope.paginationConf.totalItems = data.total;//更新总记录数
        })
    };
	
    /**
     * 定义新增方法
     */
    $scope.save = function () {
        var object = null;
        if ($scope.areas.id) {
            object = areasService.updateAreas($scope.areas);//修改
        } else {
            object = areasService.saveAreas($scope.areas);//增加
        }
        object.success(function (data) {
					  
            if (data.success) {
					//重新查询
                $scope.reloadList();//重新加载
            } else {
                alert(data.message);
            }
        });
    };

    /**
     * 定义根据id查找
     */
    $scope.findOne = function (id) {
							   
        areasService.findOne(id).success(function (data) {
            $scope.areas = data;
        })
    };
	
    /**
     * 删除方法
     * @param idList
     */
    $scope.removeAreasList = function (idList) {
        if (confirm("是否删除?")) {
            areasService.removeAreasList(idList).success(function (data) {
                if (data.success) {
							   
                    $scope.reloadList();
                } else {
                    alert(data.message);
                }
            });
        }
    };

});
